parquet-generator/src/main/java/org/apache/parquet/encoding/vectorbitpacking/BitPackingGenerator512Vector.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.encoding.vectorbitpacking;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * This class generates vector bit packers that pack the most significant bit first.
+ * The result of the generation is checked in. To regenerate the code run this class and check in the result.
+ */
+public class BitPackingGenerator512Vector {
+  private static final String CLASS_NAME_PREFIX_FOR_INT = "ByteBitPacking512Vector";
+  private static final String CLASS_NAME_PREFIX_FOR_LONG = "ByteBitPacking512VectorForLong";
+
+  public static void main(String[] args) throws Exception {
+    String basePath = args[0];
+    //TODO: Int for Big Endian
+    //generateScheme(false, true, basePath);
+
+    // Int for Little Endian
+    generateScheme(false, false, basePath);
+
+    //TODO: Long for Big Endian
+    //generateScheme(true, true, basePath);
+
+    //TODO: Long for Little Endian
+    //generateScheme(true, false, basePath);
+  }
+
+  private static void generateScheme(boolean isLong, boolean msbFirst,
```

Comment:
I do not really get how this generation works. Some description would be great.

Reply:
@gszadovszky I have added the maven profile named vector, when execute mvn clean package -P vector, the class will be run and reads ByteBitPacking512VectorLE file to generate ByteBitPacking512VectorLE.java

Reply:
So by generating this class simply copies the file form `resources` to a source dir under the name of `.java`. Why do we need code generation then?

Reply:
I would suggest to have a separate source directory for java17 that gets "activated" only in case of the related profile is activated. This generator solution is misleading to me.

To keep this part of code clean I would also suggest to include java17 and the related class compile/unit test execution in the github actions so they would be executed on the PRs. (The only difference remains is we won't ship these in our releases.)

What do you think, @wgtmac, @jiangjiguang?

Reply:
+1 for @gszadovszky 

The core idea is to isolate the new feature as an individual unit and minimize the burden of future maintenance.

Reply:
@gszadovszky @wgtmac  I add a new module named parquet-encoding-vector on directory plugins, the new module implements all code about Vector. I think it can keep code clean about java17 and  related class compile/unit test execution.

Reply:
+1 for the option 2. But why not enable it by a new profile?

Reply:
I agree with @wgtmac. Option 2 with maven profile support seems the best way to go.

Reply:
@gszadovszky @wgtmac I agree with you, but I can't solve how to exclude a module(parquet-encoding-vector) in the profile by default.
In fact, The current solution is also simply and user only go to the plugins/parquet-encoding-vector and build with maven

Reply:
Not sure whether this answer works in this case: https://stackoverflow.com/a/13383092

Reply:
@wgtmac It makes sense. I have added profile plugins.  mvn clean package -P plugins  to enable this feature.